src/com/amazon/ion/impl/IonCursorBinary.java

```diff
@@ -661,9 +668,10 @@ private void refill(long numberOfBytesToFill) {
             throwAsIonException(e);
         }
         if (numberOfBytesFilled < 0) {
-            return;
+            return numberOfBytesFilled;
         }
         limit += numberOfBytesFilled;
+        return numberOfBytesFilled;
```

Comment:
It took me a minute to understand what this was doing. Can we rearrange this a bit?
```java
        if (numberOfBytesFilled > 0) {
            limit += numberOfBytesFilled;
        }
        return numberOfBytesFilled;
```

Reply:
Done